### PR TITLE
Bump to v0.3.5 and add kitchen

### PR DESCRIPTION
**Pull request by styner32**

### CODE REVIEW

**Comment by styner32** [comment]
Sorry. It was a mistake

File: .kitchen.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+suites:
+- name: default
+  run_list:
+  - recipe[openresty::default]
+  encrypted_data_bag_secret_key_path: ~/.chef/nitrous_encrypted_data_bag_key
+  attributes:
+    nitrous:
+      environment: staging
+    datadog:
+      environment: development
```
Comment (priestjim) [inline .kitchen.yml]: These are environment specific attributes and should not be present in a generic cookbook. Can you please review these settings?

File: recipes/commons_conf.rb [hunk 2]
```diff
@@ -103,7 +104,7 @@
     path "#{node['openresty']['log_dir']}/*.log"
     enable true
     frequency 'daily'
-    rotate 7
+    rotate 2
```
Comment (priestjim) [inline recipes/commons_conf.rb]: The default installation of nginx for Ubuntu and CentOS enforces a week's worth of logs. It should be left that way.

File: recipes/commons_conf.rb [hunk 3]
```diff
@@ -96,6 +96,7 @@
 
 if node['openresty']['logrotate']
 
+  bucket = 'goboxer-nginx-logs'
```
Comment (priestjim) [inline recipes/commons_conf.rb]: Unused variable?

File: templates/default/nginx.upstart.erb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+pre-start script
+  ${DAEMON} -t
+  if [ $? -ne 0 ]; then
+    exit $?
+  fi
+  echo "[`date -u +%Y-%m-%dT%T.%3NZ`] (sys) Starting openresty" >> /var/log/openresty.log
+end script
+
+pre-stop script
+  echo "[`date -u +%Y-%m-%dT%T.%3NZ`] (sys) Stopping openresty" >> /var/log/openresty.log
```
Comment (priestjim) [inline templates/default/nginx.upstart.erb]: This log diverges from actually keeping everything in /var/log/nginx. Can you please refactor this to actually use upstart's logging capabilities?

File: templates/default/nginx.upstart.erb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+<% unless node['openresty']['service']['upstart']['foreground'] %>
+expect fork
+<% else %>
+console output
+<% end %>
+
+exec ${DAEMON} -c "${CONFIG}" -g "daemon off;"
+
+<% if node.recipe?('nginx::passenger') and not node['nginx']['upstart']['foreground'] %>
```
Comment (priestjim) [inline templates/default/nginx.upstart.erb]: This is nginx specific?

**Pull request closed without merging**